From: Liam Howlett <liam.howlett@oracle.com>
Subject: mm/mmap: fix copy_vma() new_vma check
Date: Wed, 6 Jul 2022 00:35:33 +0000

When checking new_vma limits to ensure they are not within where the copy
of the vma will be placed, ensure new_vma is not NULL.

Link: https://lkml.kernel.org/r/20220706003522.1827240-1-Liam.Howlett@oracle.com
Fixes: ded0cf440d9b (mm: remove the vma linked list)
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Reported-by: Yu Zhao <yuzhao@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/mmap.c~mm-remove-the-vma-linked-list-fix-3
+++ a/mm/mmap.c
@@ -3347,7 +3347,7 @@ struct vm_area_struct *copy_vma(struct v
 	}
 
 	new_vma = find_vma_prev(mm, addr, &prev);
-	if (new_vma->vm_start < addr + len)
+	if (new_vma && new_vma->vm_start < addr + len)
 		return NULL;	/* should never get here */
 
 	new_vma = vma_merge(mm, prev, addr, addr + len, vma->vm_flags,
_
